pkg/asset/manifests/azure/cluster.go

```diff
@@ -86,7 +86,8 @@ func GenerateClusterAssets(installConfig *installconfig.InstallConfig, clusterID
 					PrivateDNSZoneName: installConfig.Config.ClusterDomain(),
 				},
 				Vnet: capz.VnetSpec{
-					ID: installConfig.Config.Azure.VirtualNetwork,
+					ResourceGroup: installConfig.Config.Azure.NetworkResourceGroupName,
+					Name:          installConfig.Config.Azure.VirtualNetwork,
```

Comment:
Looking at https://github.com/openshift/installer/blob/master/data/data/install.openshift.io_installconfigs.yaml#L2701, it appears that VirtualNetwork is an optional config. So, is it possible that we are creating capz.VnetSpec{} with Name as " " ?
Similar comment about https://github.com/openshift/installer/blob/master/data/data/install.openshift.io_installconfigs.yaml#L2663?

Reply:
CAPZ doesn't consider it if it's left blank. Tested it and it defaults to the `-vnet` default and creates it.